identity/identity.go

```diff
@@ -0,0 +1,47 @@
+package identity
+
+import (
+	"golang.org/x/crypto/ed25519"
+)
+
+type publicIdentity struct {
+	TrustchainID []byte `json:"trustchain_id"`
+	Target       string `json:"target"`
+	Value        []byte `json:"value"`
+}
+
+type identity struct {
+	publicIdentity
+	DelegationSignature          []byte `json:"delegation_signature"`
+	EphemeralPublicSignatureKey  []byte `json:"ephemeral_public_signature_key"`
+	EphemeralPrivateSignatureKey []byte `json:"ephemeral_private_signature_key"`
+	UserSecret                   []byte `json:"user_secret"`
+}
+
+func generateIdentity(config config, userIDString string) (*identity, error) {
+	userID := hashUserID(config.TrustchainID, userIDString)
+	userSecret := createUserSecret(userID)
+
+	eprivSignKey, epubSignKey, err := ed25519.GenerateKey(nil)
```

Comment:
is it possible to import ed25519 as crypto and call `crypto.GenerateKey(nil)` ?

Reply:
Nope. This is the name of the algorithm we generate a key for